src/docker/ContextManager.ts

```diff
@@ -106,13 +116,18 @@ export class DockerContextManager implements ContextManager, Disposable {
             // Best effort
         }
 
+        // Set up a lazy to determine the compose command to use, and also set up clearing it on context change
+        this.composeCommandLazy = new AsyncLazy<string>(async () => this.determineComposeCommand());
+        this.composeCommandLazyDisposable = this.onContextChanged(() => this.composeCommandLazy.clear());
+
         // Set the initial DockerApiClient to be the context loading client
         ext.dockerClient = new ContextLoadingClient(this.loadingFinishedEmitter.event);
     }
 
     public dispose(): void {
         void this.configFileWatcher?.dispose();
         void this.contextFolderWatcher?.dispose();
+        void this.composeCommandLazyDisposable?.dispose();

```

Comment:
Maybe instead of having separate members for the file watchers and the composeCmdLazyDisposable, the ContextManager could just have one "disposables array"? These file watchers etc do not seep to be used outside of class constructor

Reply:
Good idea!